subsystems/viamserver/viamserver.go

```diff
@@ -304,15 +309,25 @@ func (s *viamServer) isRestartAllowed(ctx context.Context) (bool, error) {
 				resultChan <- nil
 				return
 			}
-			resultChan <- errors.New("viam-server reports it is unsafe to restart")
+			resultChan <- errUnsafeToRestart
 		}()
 	}
+
 	var combinedErr error
 	for i := 1; i <= len(urls); i++ {
 		result := <-resultChan
+
+		// If any test URL reports it is explicitly _safe_ to restart (nil value sent to
+		// resultChan), we can assume we can restart viam-server.
 		if result == nil {
 			return true, nil
 		}
+		// If any test URL reports it is explicitly _unsafe_ to restart (errUnsafeToRestart),
+		// we can assume we should not restart viam-server.
+		if errors.Is(err, errUnsafeToRestart) {
```

Comment:
I'm fairly certain a log like:
```
7/18/2025, 4:07:20 PM warn viam-agent   viamserver/viamserver.go:349   viam-server reports it is unsafe to restart viam-server reports it is unsafe to restart viam-server reports it is unsafe to restart viam-server reports it is unsafe to restart viam-server reports it is unsafe to restart
```
Is caused by five test URLs pointing to the same `viam-server` instance _all_ reporting that `viam-server` is unsafe to restart. I do not think we want to combine the errors in that case, and can, conservatively, return `false, errUnsafeToRestart` in the event that any one of the test URLs reports `RestartAllowed: false`.